### ISSUE TITLE
Rust loads from cache with --recompile

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#1637]
In cleaning up the compiler, I noticed while passing the `--recompile` flag with debug, that Rust will still attempt to load the local cache of a file even though it should "know" that it is not going to be used.  For example:

```
DEBUG JS - args [ "tests/005_more_imports.ts" ]
DEBUG RS - code_fetch. specifier tests/005_more_imports.ts referrer .
DEBUG RS - resolve_module specifier tests/005_more_imports.ts referrer .
DEBUG RS - module_name: /Users/kkelly/github/deno/tests/005_more_imports.ts, filename: /Users/kkelly/github/deno/tests/005_more_imports.ts
DEBUG RS - fetch local or reload /Users/kkelly/github/deno/tests/005_more_imports.ts is_module_remote false
DEBUG RS - found local source 
DEBUG RS - load_cache code: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js map: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js.map
```

**Comment by kitsonk** [comment]
Actually it is "worse" than that, because on the `code_fetch` from the compiler, it loads it a second time:

```
DEBUG JS - args [ "tests/005_more_imports.ts" ]
DEBUG RS - code_fetch. specifier tests/005_more_imports.ts referrer .
DEBUG RS - resolve_module specifier tests/005_more_imports.ts referrer .
DEBUG RS - module_name: /Users/kkelly/github/deno/tests/005_more_imports.ts, filename: /Users/kkelly/github/deno/tests/005_more_imports.ts
DEBUG RS - fetch local or reload /Users/kkelly/github/deno/tests/005_more_imports.ts is_module_remote false
DEBUG RS - found local source 
DEBUG RS - load_cache code: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js map: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js.map
DEBUG RS - >>>>> compile_sync START
...
DEBUG TS - os.ts codeFetch tests/005_more_imports.ts .
DEBUG RS - code_fetch. specifier tests/005_more_imports.ts referrer .
DEBUG RS - resolve_module specifier tests/005_more_imports.ts referrer .
DEBUG RS - module_name: /Users/kkelly/github/deno/tests/005_more_imports.ts, filename: /Users/kkelly/github/deno/tests/005_more_imports.ts
DEBUG RS - fetch local or reload /Users/kkelly/github/deno/tests/005_more_imports.ts is_module_remote false
DEBUG RS - found local source 
DEBUG RS - load_cache code: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js map: /Users/kkelly/.deno/gen/e74a8e8c25c18115805245674c5d1a615441f4b7.js.map
```

**Comment by kitsonk** [comment]
Closed in 37b05740facd26b5f20b08c3500cd782f70de546
